inc/azure_macro_utils/macro_utils.h

```diff
@@ -153,53 +181,23 @@ static const char* MU_C2(enumName,Strings)(enumName value)         \
 // this macro returns the number of enum values (taking into account that an invalid value is generated)
 #define MU_ENUM_VALUE_COUNT(...) (MU_COUNT_ARG(__VA_ARGS__) + 1)
```

Comment:
>MU_ENUM_VALUE_COUNT [](start = 8, length = 19)

doesn't seem to be used anymore in this file...

Reply:
It is public. The reason for MU_ENUM_VALUE_COUNT is that we now have enums that have the extra INVALID and enums that do not.
MU_ENUM_VALUE_COUNT should abstract the fact that we add +1 for the INVALID.
Of course we should have MU_ENUM_VALUE_COUNT_WITHOUT_INVALID also.

Which I just added.

---
In reply to: [328357652](https://github.com/Azure/azure-macro-utils-c/pull/30#discussion_r328357652) [](ancestors = 328357652)